### PR TITLE
Improve error message when launching kernels fails

### PR DESCRIPTION
**Pull request by pfultz2**
This will let the user know which targets migraphx was built for and how to build migraphx for their gpu.

### CODE REVIEW

**Review by umangyadav — APPROVED** [review]

**Review by TedThemistokleous — CHANGES_REQUESTED** [review]
add licensing / fix tidy. Otherwise LGTM

File: src/targets/gpu/device/targets.hpp.in [hunk 1]
```diff
@@ -0,0 +1,48 @@
+/*
+* The MIT License (MIT)
+*
+* Copyright (c) 2015-2022 Advanced Micro Devices, Inc. All rights reserved.
```
Comment (TedThemistokleous) [inline src/targets/gpu/device/targets.hpp.in:4]: 2023

File: src/targets/gpu/device/targets.cpp [hunk 2]
```diff
@@ -0,0 +1,66 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2015-2022 Advanced Micro Devices, Inc. All rights reserved.
```
Comment (TedThemistokleous) [inline src/targets/gpu/device/targets.cpp:4]: 2023

**Review by TedThemistokleous — APPROVED** [review]
Just fix the dates but LGTM

**Pull request merged**